packages/eas-cli/src/update/configure.ts

```diff
@@ -0,0 +1,291 @@
+  modifyConfig,
+  workflows,
+}: {
+  modifyConfig: Partial<ExpoConfig>;
+  workflows: Record<Platform, Workflow>;
+}): void {
+  Log.addNewLineIfNone();
+  Log.warn(
+    `It looks like you are using a dynamic configuration! ${learnMore(
+      'https://docs.expo.dev/workflow/configuration/#dynamic-configuration-with-appconfigjs)'
+    )}`
+  );
+  Log.warn(
+    `In order to finish configuring your project for EAS Update, you are going to need manually add the following to your app.config.js:\n${learnMore(
+      'https://expo.fyi/eas-update-config.md'
+    )}\n`
+  );
+  Log.log(chalk.bold(JSON.stringify(modifyConfig, null, 2)));
+  Log.addNewLineIfNone();
+
+  if (workflows.android === Workflow.GENERIC || workflows.ios === Workflow.GENERIC) {
+    Log.warn(
+      `You will also have to manually edit the projects ${chalk.bold(
+        'Expo.plist/AndroidManifest.xml'
+      )}. ${learnMore('https://expo.fyi/eas-update-config.md#native-configuration')}`
+    );
+  }
+
+  Log.addNewLineIfNone();
+}
+
+/**
+ * Make sure that the current `app.json` configuration for EAS Updates is set natively.
+ */
+export async function ensureEASUpdatesIsConfiguredNativelyAsync(
+  graphqlClient: ExpoGraphqlClient,
+  {
+    exp,
+    projectId,
+    projectDir,
+    platform,
+    workflows,
+  }: {
+    exp: ExpoConfig;
+    projectId: string;
+    projectDir: string;
+    platform: RequestedPlatform;
+    workflows: Record<Platform, Workflow>;
+  }
+): Promise<void> {
+  if (['all', 'android'].includes(platform) && workflows.android === Workflow.GENERIC) {
+    await syncAndroidUpdatesConfigurationAsync(graphqlClient, projectDir, exp, projectId);
+    Log.withTick(`Configured ${chalk.bold('AndroidManifest.xml')} for EAS Update`);
+  }
+
+  if (['all', 'ios'].includes(platform) && workflows.ios === Workflow.GENERIC) {
+    await syncIosUpdatesConfigurationAsync(graphqlClient, projectDir, exp, projectId);
+    Log.withTick(`Configured ${chalk.bold('Expo.plist')} for EAS Update`);
+  }
+}
+
+/**
+ * Make sure EAS Update is fully configured in the current project.
+ * This goes over a checklist and performs the following checks or changes:
+ *   - Enure the `expo-updates` package is currently installed.
+ *   - Ensure `app.json` is configured for EAS Updates
+ *     - Sets `runtimeVersion` if not set
+ *     - Sets `updates.url` if not set
+ *   - Ensure latest changes are reflected in the native config, if any
+ */
+export async function ensureEASUpdatesIsConfiguredAsync(
+  graphqlClient: ExpoGraphqlClient,
+  {
+    exp: expWithoutUpdates,
+    projectId,
+    projectDir,
+    platform,
+  }: {
+    exp: ExpoConfig;
+    projectId: string;
+    projectDir: string;
+    platform: RequestedPlatform;
+  }
+): Promise<{ projectChanged: boolean }> {
+  const hasExpoUpdates = isExpoUpdatesInstalledOrAvailable(
+    projectDir,
+    expWithoutUpdates.sdkVersion
+  );
+  if (!hasExpoUpdates) {
+    await installExpoUpdatesAsync(projectDir, { silent: true });
```

Comment:
[sanity] Toggle off silent when debug logging is enabled, makes our job easier in the future.